celi_framework/main.py

```diff
@@ -106,14 +136,65 @@ def bool_opt(opt: str, env_var: str, help: str):
 
     args = parser.parse_args()
 
+    # Always initialize tool_config at the beginning of the function
+    tool_config = {}  # Default to an empty dictionary or a suitable default
+
+    # # Resolve the configuration path
+    # config_path = resolve_config_path(args.tool_config_json) if args.tool_config_json else get_default_config_path() <----
+
+    # if args.tool_config_json:
+    #     config_path = resolve_config_path(args.tool_config_json)
+    # else:
+    #     # TODO - If TOOL_CONFIG_JSON=celi_framework/examples/wikipedia/example_config.json or if no .env file
+    #     # TODO- then we can do the following. but if TOOL_CONFIG_JSON= some other non-existant path then throw an error saying that you need to specific TOOL_CONFIG_JSON path in .env
+    #     config_path = get_default_config_path()
+    #
+    # # Attempt to read the configuration file
+    # if config_path and os.path.exists(config_path):
+    #     try:
+    #         config_data = read_json_from_file(config_path)
+    #         tool_config.update(config_data)  # Safely update tool_config with loaded data
+    #     except Exception as e:
+    #         print(f"Failed to read the configuration file: {e}")
+    # else:
+    #     print(f"Configuration file not found or path is invalid: {config_path}")
+    #
+    # # Check if tool_config is populated properly
+    # if not tool_config:
+    #     print("Error: Tool configuration data is missing or could not be loaded.")
+    #     return None
+
+    # Resolve and load the configuration
+    config_path = resolve_config_path(args.tool_config_json)
+    if not config_path:
+        print("Valid configuration file path not found.")
+        return None
+
+    try:
+        tool_config = read_json_from_file(config_path)
+        if not all(key in tool_config for key in ['example_url', 'target_url']):
```

Comment:
This validation shouldn't go here.  This is a general main script for celi, but the validation you are doing here is specific to the wikipedia example.  Other use cases would have different tool configurations.  The point of putting the config in a JSON file is because it can change entirely with the use case.  If we knew that every tool had an example_url and target_url we could have just made those additional options to this script.